File: easy/kids_with_candies/kids_with_candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        max_candies = max(candies)
        can_have_max_candies = []

        for i in range(0, len(candies)):
            candies_needed = max_candies - candies[i]
            if candies_needed <= extraCandies:
                can_have_max_candies.append(True)
            else:
                can_have_max_candies.append(False)

        return can_have_max_candies


s = Solution()
print("Solution 1 : ", s.kidsWithCandies([2, 3, 5, 1, 3], 3))
print("Solution 2 : ", s.kidsWithCandies([4, 2, 1, 1, 2], 1))
print("Solution 3 : ", s.kidsWithCandies([12, 1, 12], 10))

File: easy/array/kids_with_candies/kids_with_candies.go
package main

import "fmt"

func findMax(array []int) int {
	max := array[0]
	for i := 1; i < len(array); i++ {
		if max < array[i] {
			max = array[i]
		}
	}
	return max
}

func kidsWithCandies(candies []int, extraCandies int) []bool {
	maxCandies := findMax(candies)
	canHaveMaxCandies := make([]bool, 0, len(candies))

	for i := 0; i < len(candies); i++ {
		candiesNeeded := maxCandies - candies[i]
		if candiesNeeded <= extraCandies {
			canHaveMaxCandies = append(canHaveMaxCandies, true)
		} else {
			canHaveMaxCandies = append(canHaveMaxCandies, false)
		}
	}

	return canHaveMaxCandies
}

func main() {
	fmt.Println("Solution 1 : ", kidsWithCandies([]int{2, 3, 5, 1, 3}, 3))
	fmt.Println("Solution 2 : ", kidsWithCandies([]int{4, 2, 1, 1, 2}, 1))
	fmt.Println("Solution 1 : ", kidsWithCandies([]int{12, 1, 12}, 10))
}
